### PR TITLE
PFM-2968: Refactor: Add Pagination value to billSearchCriteria

### PR DESCRIPTION
**Pull request by harshal-egov**
PFM-2968: Refactor: Add Pagination value to billSearchCriteria

### CODE REVIEW

File: backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/util/ExpenseCalculatorUtil.java [hunk 1]
```diff
@@ -139,11 +139,11 @@ public List<Bill> fetchBills(RequestInfo requestInfo, String tenantId, String co
         List<String> billIds = expenseCalculatorRepository.getBills(contractId, tenantId);
 
         StringBuilder url = searchURI(configs.getBillHost(), configs.getExpenseBillSearchEndPoint());
-        //Pagination pagination = Pagination.builder().limit(100).offSet(0).order(Order.ASC).build();
+        Pagination pagination = Pagination.builder().limit(100).offSet(0).order(Order.ASC).build();
```
Comment (jagankumar-egov) [inline backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/util/ExpenseCalculatorUtil.java:142]: keep a pagination object in search criteria, and change the offset and limit to a environment variable

Comment (harshal-egov) [reply backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/util/ExpenseCalculatorUtil.java]: Changes implemented in new commits.

File: backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/service/ExpenseCalculatorService.java [hunk 2]
```diff
@@ -263,11 +264,25 @@ private void persistMeta(List<Bill> bills,Map<String, String> metaInfo) {
...
+
         RequestInfo requestInfo=calculatorSearchRequest.getRequestInfo();
+
         String tenantId=calculatorSearchRequest.getSearchCriteria().getTenantId();
 
         Map<String,BillMapper> billMappers=expenseCalculatorRepository.getBillMappers(calculatorSearchRequest);
-        List<Bill> bills=expenseCalculatorUtil.fetchBillsWithBillIds(requestInfo,tenantId,new ArrayList<>(billMappers.keySet()));
+
+        //set total count
+        calculatorSearchRequest.getPagination().setTotalCount(billMappers.size());
```
Comment (kavi-egov) [inline backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/service/ExpenseCalculatorService.java:290]: don't do this, this should be the result from a separate count query ran for the same query with no pagination. remove the bills.size

File: backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/web/models/CalculatorSearchRequest.java [hunk 3]
```diff
@@ -27,4 +27,7 @@ public class CalculatorSearchRequest {
 	@JsonProperty("searchCriteria")
 	private CalculatorSearchCriteria searchCriteria;
 
+	@JsonProperty("pagination")
+	private Pagination pagination;
```
Comment (kavi-egov) [inline backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/web/models/CalculatorSearchRequest.java:31]: @notnull for all elements

**Review by kavi-egov — CHANGES_REQUESTED** [review]

**Review by vishalgupta-eGov — APPROVED** [review]

**Review by subhashini-egov — APPROVED** [review]

File: backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/service/ExpenseCalculatorService.java (hunk 2 above)
Comment (harshal-egov) [reply backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/service/ExpenseCalculatorService.java]: Change implemented in later commits.

File: backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/web/models/CalculatorSearchRequest.java (hunk 3 above)
Comment (harshal-egov) [reply backend/expense-calculator/src/main/java/org/egov/digit/expense/calculator/web/models/CalculatorSearchRequest.java:35]: Change implemented in later commits.

**Review by kavi-egov — APPROVED** [review]

**Pull request merged**